pages/_app.js

```diff
@@ -74,13 +83,22 @@ export default function App({ Component, pageProps }) {
       <main>
         <Component
           {...pageProps}
+          randomActivity={randomActivity}
           activityData={activityData}
           favoriteActivitiesList={favoriteActivitiesList}
           onAddActivity={handleAddActivity}
           onEditActivity={handleEditActivity}
           onDeleteActivity={handleDeleteActivity}
           onToggleFavorite={handleToggleFavorite}
         />
+        <button
+          type="button"
+          onClick={() => {
+            getRandomActivity(), router.push("/spotlight");
```

Comment:
```suggestion
            getRandomActivity();
            router.push("/spotlight");
```
separating two actions with a comma is a bit unusual, please use just 2 lines instead.